### PR TITLE
Improvements for K8S deployment (especially ITs)

### PR DESCRIPTION
**Pull request by almahmoud**
1) Adds GHCR build, to get a k8s container image built and pushed to `ghcr.io` under repository owner, with no credentials needed (complementing the DockerHub/Quay.io pushed that are only for `galaxyproject` repo and require credentials). Demonstrated for the Bioconductor fork (https://github.com/Bioconductor/galaxy/pkgs/container/galaxy), and could also be useful for AnVIL fork where it could replace the custom workflow that is currently being used.

2) Updates `pykube-ng` version to latest available, making the deployment compatible with newer versions of Kubernetes (would error out on an API call for ingresses otherwise). It seems `pykube-ng` is not being updated anymore though, and there is an official Kubernetes API client for python now, which should likely be the long-term solution here imo.

3) Add ability to override the TLS secret for ITs' ingress resources, notably useful for easily setting up a wildcard cert instead of relying on `cert-manager` to issue a new certificate per subdomain (which eventually hits the rate limit at least for letsencrypt).

4) Adds ability to override ingress version, currently useful for running on old kubernetes versions (and added back conditional logic to work with old `v1beta1` API version).

I don't remember ITs on Kubernetes being tested, so I didn't really look into adding tests. The changes aren't too complicated though. One thing I did notice is that although default is set [here](https://github.com/galaxyproject/galaxy/compare/dev...Bioconductor:galaxy:dev?expand=1#diff-4b73c7a6546e8d68305e5e776cf411dc3e5db8a53c980a174ee330eaa22a1583R114), `runner_params.get()` still returns `None` when it's not set. Is that a bug in the runner_params parsing?

cc: @nuwang @ksuderman @afgane 

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by ksuderman** [comment]
I am testing this now.

**Comment by ksuderman** [comment]
I haven't had a chance to look into it, but there does seem to be a bug in the way the `runner_params` are parsed.  For testing I worked around the problem by providing defaults to the `get` methods [here](https://github.com/Bioconductor/galaxy/blob/d3717adfe23461f4ecf4dca314f91b3dd8836e68/lib/galaxy/jobs/runners/kubernetes.py#L255) and [here](https://github.com/Bioconductor/galaxy/blob/d3717adfe23461f4ecf4dca314f91b3dd8836e68/lib/galaxy/jobs/runners/kubernetes.py#L438), but that simply masks the underlying problem.  However, until that problem is fixed we may want to consider including the work around in the PR otherwise we just get unhandled `KeyError`s.

**Review by ksuderman — APPROVED** [review]
As mentioned, we may want to include a work around to the `runner_params` parsing problem until that has been resolved.

File: lib/galaxy/jobs/runners/kubernetes.py [hunk 1]
```diff
@@ -249,6 +252,7 @@ def __configure_port_routing(self, ajs):
         service = Service(self._pykube_api, k8s_service_obj)
         service.create()
         ingress = Ingress(self._pykube_api, k8s_ingress_obj)
+        ingress.version = self.runner_params.get("k8s_ingress_api_version")
```
Comment (nuwang) [inline lib/galaxy/jobs/runners/kubernetes.py]: Try using
```suggestion
        ingress.version = self.runner_params["k8s_ingress_api_version"]
```
That should cause the default to be returned I think. Runner_params is  custom class that overrides subscripting behaviour.

**Review by nuwang — APPROVED** [review]
lgtm! I think the default param issue can be fixed by directly subscripting instead of using dict.get()

**Comment by almahmoud** [comment]
I added another [commit](https://github.com/galaxyproject/galaxy/pull/18514/commits/b42b416a27ef1f5b77d2fa3e3167dd796d0d66bd) that makes jobs in kubernetes not be marked as running until the pod is actually running, as opposed to for eg waiting for container image to be pulled. I also changed the `.active` for IT entrypoints to look for whether the job is running as opposed to just not finished, which fixes the "FIXME" comment about queued being detected as active in some cases

File: lib/galaxy/jobs/runners/kubernetes.py [hunk 2]
```diff
@@ -767,10 +803,12 @@ def check_watched_item(self, job_state):
                                 pass
                         else:
                             pass
-                    else:
+                    elif self.__check_job_pod_running(job_state):
                         log.debug("Job set to running...")
                         job_state.running = True
                         job_state.job_wrapper.change_state(model.Job.states.RUNNING)
+                    else:
+                        pass
```
Comment (nuwang) [inline lib/galaxy/jobs/runners/kubernetes.py]: I think a debug statement would be helpful here. Maybe something like this?:
```suggestion
                    else:
                        log.debug(f"Job id: {job_state.job_id} with k8s id: {job.name} scheduled and is waiting to start...")
```

File: lib/galaxy/model/__init__.py [hunk 3]
```diff
@@ -2912,8 +2912,7 @@ def __init__(
     @property
     def active(self):
         if self.configured and not self.deleted:
-            # FIXME: don't included queued?
-            return not self.job.finished
+            return self.job.running
```
Comment (nuwang) [inline lib/galaxy/model/__init__.py:2915]: This makes sense I think, I just wonder whether this should be its own PR?

Comment (almahmoud) [reply lib/galaxy/model/__init__.py:2915]: Happy to remove the last few commits then and add them as separate PR

File: lib/galaxy/jobs/runners/kubernetes.py (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy/jobs/runners/kubernetes.py]: I think info is the right level for that, it's a useful breadcrumb in the context of errors.

**Review by nuwang — APPROVED** [review]

File: lib/galaxy/model/__init__.py (hunk 3 above)
Comment (nuwang) [reply lib/galaxy/model/__init__.py:2915]: Other than running, all other states should probably return inactive, so this change makes sense. Let's just keep it.
